File: Arrays/Search & Sort/majority-element.ts
// Majority Element
// http://www.geeksforgeeks.org/majority-element/

/**
 * Author: Arunkumar Srisailapathi
 * Email: sail.arunkumar@gmail.com
 * Web: arunkumars08.github.io
 */

/**
 * Input: [2,2,2,2,5,5,2,3,3]
 * Output: 2
 */

class MajorityElement {

    find (arr: Array<number>): any {
        let hash: any = {};
        const len: number = arr.length;
        for (let i = 0; i < len; ++ i) {
            let a: number = arr[i];
            if (hash.hasOwnProperty(a)) {
                hash[a] = hash[a] + 1;
                if (hash[a] > (len / 2)) {
                    return a;
                }
            } else {
                hash[a] = 1;
            }
        }
        return -1;
    }

    constructor (arr: Array<number>) {
        console.log(this.find(arr));
    }
}

File: Arrays/Search & Sort/majority-element.py
# Majority Element
# http://www.geeksforgeeks.org/majority-element/
# Author: Srivathsan
# Email: srivathsankr7@gmail.com

def majority_element(arr):
    n = len(arr)
    hashmap = {}
    for i in arr:
        if i in hashmap:
            val = hashmap[i] + 1
            if val > n/2:
                return i
            else:
                hashmap[i] = val
        else:
            hashmap[i] = 1
    return -1

if __name__ == '__main__':
    arr = [2,2,2,2,5,5,2,3,3]
    print(majority_element(arr))